compiler/rustc_parse/src/lexer/unescape_error_reporting.rs

```diff
@@ -95,7 +95,7 @@ pub(crate) fn emit_unescape_error(
                     }
                     escaped.push(c);
                 }
-                if escaped.len() != lit.len() {
+                if escaped.len() != lit.len() || full_lit_span.is_empty() {
```

Comment:
I understand that this prevents the negative overflow & that makes sense but why wasn't this an issue before? Don't have the time to dig deep into the code myself.

Reply:
Before we were replacing the full span with the new string (instead of doing `BytePos` math). We just end up with an invalid suggestion that won't be displayed.